controllers/update_status.go

```diff
@@ -624,9 +640,77 @@ func validateProcessGroup(ctx context.Context, r *FoundationDBClusterReconciler,
 	processGroupStatus.UpdateCondition(fdbv1beta2.PodFailing, failing, cluster.Status.ProcessGroups, processGroupStatus.ProcessGroupID)
 	processGroupStatus.UpdateCondition(fdbv1beta2.PodPending, false, cluster.Status.ProcessGroups, processGroupStatus.ProcessGroupID)
 
+	disableTaintFeature := false
+	for _, taintConfiguredKey := range cluster.Spec.AutomationOptions.Replacements.TaintReplacementOptions {
+		if *taintConfiguredKey.Key == "*" && *taintConfiguredKey.DurationInSeconds < 0 {
+			// Disable detecting or replacing taint key
+			disableTaintFeature = true
+			logger.Info("Disable taint feature", "Disabled", disableTaintFeature)
+			break
+		}
+	}
+
+	if !disableTaintFeature {
+		// Update taint status
+		updateTaintCondition(ctx, r, cluster, pod, processGroupStatus)
+	}
+
 	return nil
 }
 
+// updateTaintCondition checks pod's node taint label and update pod's taint-related condition accordingly
+func updateTaintCondition(ctx context.Context, r *FoundationDBClusterReconciler, cluster *fdbv1beta2.FoundationDBCluster,
+	pod *corev1.Pod, processGroupStatus *fdbv1beta2.ProcessGroupStatus) error {
+	logger := log.WithValues("namespace", cluster.Namespace, "cluster", cluster.Name, "reconciler", "updateStatus", "step", "updateTaintCondition")
+	node := &corev1.Node{}
+	err := r.Get(ctx, client.ObjectKey{Name: pod.Spec.NodeName}, node)
+	if err != nil {
+		log.Info("Get pod's node fails", "Pod", pod.Name, "Pod's node name", pod.Spec.NodeName, "err", err)
+	} else {
+		// Check the tainted duration and only mark the process group tainted after the configured tainted duration
+		// Future: Do not replace Pod that tolerates a taint
```

Comment:
Can we make this an `TODO` and create a GitHub issue for that (and reference it here)?

Reply:
https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1583